src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,118 @@
+  const dispatch = useAppDispatch();
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">5</td>
-            <td className="is-vcentered"> </td>
+  const query = useAppSelector(state => state.filter.query);
+  const status = useAppSelector(state => state.filter.status);
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">laboriosam mollitia et enim quasi adipisci quia provident illum</p>
-            </td>
+  const preparedList = useMemo(() => {
+    let newList: Todo[] = [...todos];
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">6</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">qui ullam ratione quibusdam voluptatem quia omnis</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">7</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">illo expedita consequatur quia in</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+    if (query) {
+      newList = newList.filter((todo: Todo) => {
+        return todo
+          .title
+          .toLowerCase()
+          .includes(query.toLowerCase());
+      });
+    }
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">8</td>
+    if (status === 'completed') {
+      return newList.filter((todo: Todo) => todo.completed);
+    }
 
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
+    if (status === 'active') {
+      return newList.filter((todo: Todo) => !todo.completed);
+    }
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">quo adipisci enim quam ut ab</p>
-            </td>
+    return newList;
+  }, [query, status]);
```

Comment:
<img width="983" alt="Screenshot 2024-01-12 at 15 57 44" src="https://github.com/mate-academy/react_redux-list-of-todos/assets/135232001/53c8c4f8-7831-4df7-9a74-fda1b8a2156f">